src/decaf.cpp

```diff
@@ -195,6 +195,38 @@ namespace decaf {
     return ExecutionResult::Continue;
   }
 
+  ExecutionResult Interpreter::visitSDiv(llvm::BinaryOperator& op) {
+    StackFrame& frame = ctx->stack_top();
+
+    auto lhs = normalize_to_int(frame.lookup(op.getOperand(0), *z3));
+    auto rhs = normalize_to_int(frame.lookup(op.getOperand(1), *z3));
+
+    if (ctx->check(rhs == 0) == z3::sat) {
+      queue->add_failure(ctx->solver.get_model());
+    }
+    ctx->add(rhs != 0);
+
```

Comment:
Looking at this a second time we're also missing a second edge case here. We need another check for the case where `rhs == INT_MIN` and `lhs == -1`. You should be able to use `z3::bvsdiv_no_overflow` to check for this.